zebedee-cms/src/main/java/com/github/onsdigital/zebedee/model/approval/ApproveTask.java

```diff
@@ -526,16 +526,17 @@ protected String incrementDatasetVersionByOne(String datasetVersion) {
         return letter + Integer.toString(number);
     }
 
-    protected String findCorrectDatasetVersion(List<String> listOfUris) {
+    protected String findCorrectDatasetVersion(List<String> listOfUris) throws IllegalArgumentException {
```

Comment:
so this exception isn't getting caught - so won't this just cause the application to fall over? I may be misremembering my java.

Reply:
Sorry Linden we made a mistake. The IllegalArgumentException that this method throws is an Unchecked Exception so we don't have to declare it. So we removed the throws clause in the method signature. And also the exception that this method throws will be caught further up in the function hierarchy, namely on this catch clause
[https://github.com/ONSdigital/zebedee/blob/387c589255f632e9c3a0a3330b0ae8c69302e50[…]a/com/github/onsdigital/zebedee/model/approval/ApproveTask.java](https://github.com/ONSdigital/zebedee/blob/387c589255f632e9c3a0a3330b0ae8c69302e505/zebedee-cms/src/main/java/com/github/onsdigital/zebedee/model/approval/ApproveTask.java#L184)

Reply:
Great, thanks Dionisi